.github/workflows/build.yml

```diff
@@ -0,0 +1,25 @@
+name: build
+
+on:
+  push:
+    branches:
+      - main
+      - release/**
+
+jobs:
+  dist:
+    name: Build wheel and .tar files for release
+    timeout-minutes: 10
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: 3.12
+      - run: |
+          pip install build
+          python -m build
```

Comment:
is this dependency needed? what is it doing?

Reply:
my understanding is this is what's building the whl & tar files in the artifact, I based this off of 
https://packaging.python.org/en/latest/tutorials/packaging-projects/#generating-distribution-archives